### PR TITLE
cson parser returns `any` rather than `Object`

### PR DESCRIPTION
**Pull request by dex4er**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

CSON parsers returns non-Object too:

```js
const result = require('cson').parse('"foo"')
assert(! result instanceof Object)
```

so basically it should return and accept `any` rather than `Object`.

### CODE REVIEW

**Comment by stpettersens** [comment]
👎 

Thank you for the commit, but I do not agree with the changes. 

This  definition was written for the CSON module at https://www.npmjs.com/package/cson, which according to the documentation should return only return `string`s or `Objects` on its methods.

I'm open to have this reviewed by other contributors on here before ruling out your commit though.

It's possible it should be changed if this module is now being used across different CSON implementations, but I wish to stick to @bevry 's.

Maybe @balupton (of @bevry and I believe the original author of that module) has some thoughts on this?

**Comment by DanielRosenwasser** [comment]
Would a more accurate type be something like `string | { [prop: string]: any }`?

**Comment by balupton** [comment]
Delegating my opinion over to @jkrems from https://github.com/groupon/cson-parser

**Comment by dex4er** [comment]
@DanielRosenwasser what about `parse('true')` or `parse('666')` or `parse('null')`?

So basically this is really `any` type... At least any of JSON types.

**Comment by jkrems** [comment]
`CSON.parse` is just like `JSON.parse` and will generally return the same kinds of values. So it will return something along the lines of...

```ts
type ParseResult = string | number | boolean | null | ParseResult[] | { [string]: ParseResult };

function deserialize(data: string): ParseResult {
  return CSON.parse(data);
}
```

**Comment by jkrems** [comment]
P.S.: There's very limited meaningful type checking on a generic deserialization method. I would assume that somebody would write something similar to this in practice maybe?

```ts
function deserialize<T : ParseResult>(data: string): T {
  return CSON.parse(data);
}
```

Problem is that I just discovered that Typescript doesn't seem to allow circular references inside of type defs..? Not sure what the official solution is. The above definitely works in flow.

**Comment by jkrems** [comment]
Guess the fact that `JSON.parse` is untypable in Typescript is a known bug? https://github.com/Microsoft/TypeScript/issues/6230#issuecomment-167023181

**Comment by jkrems** [comment]
Given that Typescript itself types `JSON.parse` as returning `any`, I'm pretty convinced that `any` is the appropriate return type for `CSON.parse` as well. It would be consistent with the host language:

```ts
    /**
      * Converts a JavaScript Object Notation (JSON) string into an object.
      * @param text A valid JSON string.
      * @param reviver A function that transforms the results. This function is called for each member of the object.
      * If a member contains nested objects, the nested objects are transformed before the parent object is.
      */
    parse(text: string, reviver?: (key: any, value: any) => any): any;
```

See: https://github.com/Microsoft/TypeScript/blob/16efae24338f09c9059c8ec1525166f521110935/lib/lib.d.ts#L952-L974
